File: rust/src/models/resolved_entitlement.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{feature_ref::FeatureRef, resolved_entitlement_value::ResolvedEntitlementValue};

/// Merged entitlement value for a feature across the priority hierarchy, without usage data.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ResolvedEntitlement {
    pub feature: FeatureRef,

    pub value: ResolvedEntitlementValue,
}

impl ResolvedEntitlement {
    pub fn new(feature: FeatureRef, value: ResolvedEntitlementValue) -> Self {
        Self { feature, value }
    }
}

File: typescript/src/models/resolvedEntitlement.ts
// this file is @generated
import { type FeatureRef, FeatureRefSerializer } from "./featureRef";
import {
  type ResolvedEntitlementValue,
  ResolvedEntitlementValueSerializer,
} from "./resolvedEntitlementValue";
/** Merged entitlement value for a feature across the priority hierarchy, without usage data. */
export interface ResolvedEntitlement {
  feature: FeatureRef;

  value: ResolvedEntitlementValue;
}

export const ResolvedEntitlementSerializer = {
  _fromJsonObject(object: any): ResolvedEntitlement {
    return {
      feature: FeatureRefSerializer._fromJsonObject(object["feature"]),
      value: ResolvedEntitlementValueSerializer._fromJsonObject(object["value"]),
    };
  },

  _toJsonObject(self: ResolvedEntitlement): any {
    return {
      feature: FeatureRefSerializer._toJsonObject(self.feature),
      value: ResolvedEntitlementValueSerializer._toJsonObject(self.value),
    };
  },
};
